spring-integration-core/src/main/java/org/springframework/integration/endpoint/SourcePollingChannelAdapter.java

```diff
@@ -129,6 +133,13 @@ protected boolean isReceiveOnlyAdvice(Advice advice) {
 	@Override
 	protected void applyReceiveOnlyAdviceChain(Collection<Advice> chain) {
 		if (AopUtils.isAopProxy(this.source)) {
+			Advisor[] advisors = ((Advised) this.source).getAdvisors();
+			for (Advisor advisor : advisors)  {
+				Advice advice = advisor.getAdvice();
+				if (advice != null && this.appliedAdvices.contains(advice)) {
+					((Advised) this.source).removeAdvice(advice);
```

Comment:
See the linked JIRA and its SO question - the reason we moved creating the poller to start() from onInit() was because it prevented adding extra advices in test cases.

Also, the advice chain could be changed between stops so this allows that.